activerecord/lib/active_record/connection_adapters/postgresql/oid/array.rb

```diff
@@ -33,7 +33,7 @@ def deserialize(value)
 
           def cast(value)
             if value.is_a?(::String)
-              value = @pg_decoder.decode(value)
+              value = @pg_decoder.decode(value) rescue []
```

Comment:
the inline rescue there is a bit worrying, why is it needed?

Reply:
This method receives user inputs and cast it to Array. They can be malformed and that will be raised in pg-2.0. See [this commit](https://bitbucket.org/ged/ruby-pg/commits/1b081326b346368e70c9c03ee7080e28d6b3a3dc). The rescue is not there for values retrieved from the database.

Reply:
maybe we explicitly rescue TypeError then with a tiny comment about it?

```
value = begin
    @pg_decoder.decode(value)
  rescue TypeError
     # malformed array string is treated as [], will raise in PG 2.0 gem
     # this keeps a consistent implementation 
     []
   end
```

I do wonder if we should just let this error float back to the user cause they are sending in something PG can not handle so its surprising to have input eaten.